sdcm/remote.py

```diff
@@ -122,13 +122,35 @@ def _make_ssh_command(user="root", port=22, opts='', hosts_file='/dev/null',  #
 
 
 class CommandRunner:
+    _params = None
+
     def __init__(self, hostname, user='root', password=''):
         self.hostname = hostname
         self.user = user
         self.password = password
         self.log = SDCMAdapter(LOGGER, extra={'prefix': str(self)})
         self.connection = self._create_connection()
 
+    def get_instance_parameters(self):
+        """
+        Return instance parameters required to rebuild instance
+        """
+        return self._params
+
+    def _save_instance_parameters(self, params: dict):
+        tmp = {'__class__': self.__class__}
+        tmp.update(params)
+        self._params = tmp
+
+    @staticmethod
+    def build_from_parameters(params: dict):
```

Comment:
what do you mean rebuild instance? do you want to create a "clone" of the Remoter object? If so what is the logic behind it? if you need totally new connection you use `new_session=True` for the remoter.run method. This guarantees you independent SSH session... In addition to that cloning Remoter will hurt us in performance...